### PR TITLE
feat(AppBar): Move AppBar into gamut

### PR DESCRIPTION
**Pull request by borisonr**
## Title

Description
https://codecademy.atlassian.net/browse/WEB-785
---

## Merging your changes

When you are ready to merge to master and publish your changes, use the "squash and merge" button in GitHub, and follow the [PR Title Guide](https://github.com/Codecademy/client-modules#pr-title-guide) to format your PR title and write your PR merge description.

**IMPORTANT:** If your PR contains breaking changes, please remember to follow the instructions for breaking changes!

### CODE REVIEW

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/brand-components@0.8.9-alpha.ce3f10.0<br/>@codecademy/gamut-templates@2.5.8-alpha.ce3f10.0<br/>@codecademy/gamut@2.15.1-alpha.ce3f10.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://5e8b644f6ee657a2fc9fe085--codecademy-styleguide.netlify.com>https://5e8b644f6ee657a2fc9fe085--codecademy-styleguide.netlify.com</a></p><p><a href=https://app.netlify.com/sites/codecademy-styleguide/deploys/5e8b644f6ee657a2fc9fe085>Deploy Logs</a></p>

File: packages/gamut/src/AppBar/AppBarSection.tsx [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+
+class AppBarSection extends React.Component<AppBarSectionProps> {
+  render() {
+    const classes = cx({
+      [styles.section]: true,
+      [styles.sectionRight]: this.props.position === 'right',
+      [styles.sectionLeft]: this.props.position === 'left',
+      [styles.sectionCenter]: this.props.position === 'center',
+      [`${this.props.className}`]: !!this.props.className,
+    });
```
Comment (JoshuaKGoldberg) [inline packages/gamut/src/AppBar/AppBarSection.tsx:18]: Fun fact, `classnames`/`cx` does truthiness checking for us ✨ 

```suggestion
    },
    this.props.className);
```

File: packages/gamut/src/AppBar/index.tsx [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+import AppBarTab from './AppBarTab';
+import styles from './styles/index.scss';
+
+export type AppBarProps = {
+  className?: string;
+  wide?: boolean;
+};
+
+export class AppBar extends React.Component<AppBarProps> {
+  render() {
```
Comment (JoshuaKGoldberg) [inline packages/gamut/src/AppBar/index.tsx:15]: I'm guessing these are copied from the monolith - if you have the time, would you mind converting them to function components? We've been generally aligning on FCs as a style preference _(yay, hooks!)_.

File: packages/gamut/src/AppBar/AppBarTab.tsx [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+
+export type AppBarTabProps = {
+  button?: boolean;
+  className?: string;
+};
+
+class AppBarTab extends React.Component<AppBarTabProps> {
+  getInitialProps = () => ({
+    button: false,
+  });
```
Comment (JoshuaKGoldberg) [inline packages/gamut/src/AppBar/AppBarTab.tsx:13]: I think you can kill this?

```suggestion
```

**Comment by borisonr** [comment]
haha thanks @JoshuaKGoldberg yeah i probably should've opened as a draft but wasn't sure if the alpha url would still be created, but appreciate the feedback!

File: packages/gamut/src/AppBar/index.tsx (hunk 2 above)
Comment (borisonr) [reply packages/gamut/src/AppBar/index.tsx:15]: yep copied, and time is subjective right?

Comment (JoshuaKGoldberg) [reply packages/gamut/src/AppBar/index.tsx:15]: assuming you believe in it

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/brand-components@0.8.9-alpha.be99cf.0<br/>@codecademy/gamut-templates@2.5.8-alpha.be99cf.0<br/>@codecademy/gamut@2.15.1-alpha.be99cf.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://5e8b6df2e46abc57d93bb0a4--codecademy-styleguide.netlify.com>https://5e8b6df2e46abc57d93bb0a4--codecademy-styleguide.netlify.com</a></p><p><a href=https://app.netlify.com/sites/codecademy-styleguide/deploys/5e8b6df2e46abc57d93bb0a4>Deploy Logs</a></p>

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/brand-components@0.8.9-alpha.0666ff.0<br/>@codecademy/gamut-templates@2.5.8-alpha.0666ff.0<br/>@codecademy/gamut@2.15.1-alpha.0666ff.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://5e8b75b581802bb3e3996f02--codecademy-styleguide.netlify.com>https://5e8b75b581802bb3e3996f02--codecademy-styleguide.netlify.com</a></p><p><a href=https://app.netlify.com/sites/codecademy-styleguide/deploys/5e8b75b581802bb3e3996f02>Deploy Logs</a></p>

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/brand-components@0.8.9-alpha.f98e3a.0<br/>@codecademy/gamut-templates@2.5.8-alpha.f98e3a.0<br/>@codecademy/gamut@2.15.1-alpha.f98e3a.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://5e8b7a2e3489e46a4b6922c2--codecademy-styleguide.netlify.com>https://5e8b7a2e3489e46a4b6922c2--codecademy-styleguide.netlify.com</a></p><p><a href=https://app.netlify.com/sites/codecademy-styleguide/deploys/5e8b7a2e3489e46a4b6922c2>Deploy Logs</a></p>

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/brand-components@0.8.9-alpha.9671c2.0<br/>@codecademy/gamut-templates@2.5.8-alpha.9671c2.0<br/>@codecademy/gamut@2.15.1-alpha.9671c2.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://5e8b7bd6f0c9a5161fa03d45--codecademy-styleguide.netlify.com>https://5e8b7bd6f0c9a5161fa03d45--codecademy-styleguide.netlify.com</a></p><p><a href=https://app.netlify.com/sites/codecademy-styleguide/deploys/5e8b7bd6f0c9a5161fa03d45>Deploy Logs</a></p>

File: packages/gamut/src/AppBar/index.tsx [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+      <ContentContainer className={styles.contentWrapper} wide={wide}>
+        {children}
+      </ContentContainer>
+    </div>
+  );
+};
+
+export default AppBar;
+
+export { AppBarSection, AppBarTab };
```
[the remark this answers is not in the document]
Comment (jakemhiller) [reply packages/gamut/src/AppBar/index.tsx:32]: and just `export *` from each component individually

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/brand-components@0.8.9-alpha.bc206e.0<br/>@codecademy/gamut-templates@2.6.1-alpha.bc206e.0<br/>@codecademy/gamut@2.15.1-alpha.bc206e.0<br/></pre>

File: packages/styleguide/stories/AppBar.stories.js [hunk 5]
```diff
@@ -0,0 +1,38 @@
+import { AppBar, AppBarSection, AppBarTab, Logo } from '../../gamut/src';
+import React from 'react';
+import styles from './AppBar-story.scss';
+
+export default {
+  component: AppBar,
+  title: 'Component/AppBar',
+};
+
+export const defaultAppBarFullWidth = () => (
```
[the remark this answers is not in the document]
Comment (JoshuaKGoldberg) [reply packages/styleguide/stories/AppBar.stories.js:10]: Hmm, maybe something like this? https://github.com/Codecademy/client-modules/pull/695/files#diff-e9069973f86dcc68db0ee5abdead2434

In existing codeplaces, `ProgressBar` is kinda ok: https://styleguide.codecademy.com/storybook/?path=/story/component-progressbar--progressbar

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://5e8b962d8ba8c37620d0cca1--codecademy-styleguide.netlify.com>https://5e8b962d8ba8c37620d0cca1--codecademy-styleguide.netlify.com</a></p><p><a href=https://app.netlify.com/sites/codecademy-styleguide/deploys/5e8b962d8ba8c37620d0cca1>Deploy Logs</a></p>

File: packages/styleguide/stories/AppBar.stories.js (hunk 5 above)
[the remark this answers is not in the document]
Comment (borisonr) [reply packages/styleguide/stories/AppBar.stories.js:10]: cool thanks! let me know if the bit i added suffices

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/brand-components@0.8.9-alpha.cf0cd0.0<br/>@codecademy/gamut-templates@2.6.1-alpha.cf0cd0.0<br/>@codecademy/gamut@2.15.1-alpha.cf0cd0.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://5e8ba169abcfa60dec18049a--codecademy-styleguide.netlify.com>https://5e8ba169abcfa60dec18049a--codecademy-styleguide.netlify.com</a></p><p><a href=https://app.netlify.com/sites/codecademy-styleguide/deploys/5e8ba169abcfa60dec18049a>Deploy Logs</a></p>

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/brand-components@0.8.9-alpha.8e09e2.0<br/>@codecademy/gamut-templates@2.6.1-alpha.8e09e2.0<br/>@codecademy/gamut@2.15.1-alpha.8e09e2.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://5e8ba3d4f3780c24de4bb97d--codecademy-styleguide.netlify.com>https://5e8ba3d4f3780c24de4bb97d--codecademy-styleguide.netlify.com</a></p><p><a href=https://app.netlify.com/sites/codecademy-styleguide/deploys/5e8ba3d4f3780c24de4bb97d>Deploy Logs</a></p>

File: packages/styleguide/stories/AppBar.stories.js (hunk 5 above)
[the remark this answers is not in the document]
Comment (JoshuaKGoldberg) [reply packages/styleguide/stories/AppBar.stories.js:10]: This is better worded than mine 😄 🚀

[the remark this answers is not in the document]
Comment (borisonr) [reply packages/styleguide/stories/AppBar.stories.js:10]: Gotta use my English degree for something ¯\_(ツ)_/¯

**Pull request merged**